File: E-Olymp/twoCircles.kt
import java.util.*
import kotlin.math.*

fun calc_dist(x1: Double, y1: Double, x2: Double, y2: Double): Double {
        var dist: Double = sqrt((x1 - x2).pow(2) + (y1 - y2).pow(2))
        return dist 
}

fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        //circle 1
        var x1 = sc.nextDouble()
        var y1 = sc.nextDouble()
        var r1 = sc.nextDouble()
        //circle 2
        var x2 = sc.nextDouble()
        var y2 = sc.nextDouble()
        var r2 = sc.nextDouble()

        if (x1 == x2 && y1 == y2) {
                if (r1 == r2) {
                        println(-1)
                } else println(0)
        } else {
                var centre_dist: Double = calc_dist(x1, y1, x2, y2)
                if (centre_dist == r1 + r2)
                        println(1)
                else if (centre_dist > r1 +  r2)
                        println(0)
                else println(2)
        }
}

File: E-Olymp/twoCircles.cpp
#include <bits/stdc++.h>
using namespace std;

double calc_dist(double x1, double x2, double y1, double y2) {
        double dist = (double)(pow(x1 - x2, 2) + pow(y1 - y2, 2));
        return dist;
}
int main() {
        double x1, y1, r1, x2, y2, r2;
        cin >> x1 >> y1 >> r1 >> x2 >> y2 >> r2;

        if (x1 == x2 && y1 == y2) {
                if (r1 == r2) cout << -1 << endl;
                else cout << 0 << endl;
        } else {
                double centre_dist = calc_dist(x1, x2, y1, y2);
                double rad_dist = pow(r1 + r2, 2);
                if (centre_dist == rad_dist) 
                        cout << 1 << endl;
                else if (centre_dist < rad_dist)
                        cout << 2 << endl;
                else cout << 0 << endl;
        }
        return 0;
}